ERCS/erc-7644.md

```diff
@@ -0,0 +1,200 @@
+---
+eip: 7644
+title: ERC-721 Name Registry Extension
+description: Add a time-limited unique naming registry mechanism to ERC-721 tokens.
+author: Chen Liaoyuan (@chenly)
+discussions-to: https://ethereum-magicians.org/t/erc-7644-erc-721-name-registry-extension/19022
+status: Draft
+type: Standards Track
+category: ERC
+created: 2024-03-01
+requires: 721
+---
+
+## Abstract
+
+This extension defines an interface that adds a naming mechanism to [ERC-721](./eip-721.md) tokens. It allows each token to have a unique name with a set expiration date. The interface includes functions for assigning, updating, and querying names and their associated tokens, ensuring that names remain unique until they expire.
+
+## Specification
+
+The key words "MUST", "MUST NOT", "REQUIRED", "SHALL", "SHALL NOT", "SHOULD", "SHOULD NOT", "RECOMMENDED", "NOT RECOMMENDED", "MAY", and "OPTIONAL" in this document are to be interpreted as described in RFC 2119 and RFC 8174.
+
+Implementers of this extension **MUST** have all of the following functions:
+
+```solidity
+pragma solidity ^0.8.0;
+
+/**
+ * @title INameRegistry
+ * @dev Interface for the NameRegistry smart contract.
+ * This interface allows interaction with a NameRegistry, 
+ * enabling the registration, management, and lookup of names 
+ * with associated expiry dates tied to specific tokens.
+ */
+interface INameRegistry {
```

Comment:
It's unclear if this is supposed to be implemented by the same contract implementing `ERC721` or if this is a completely separate contract.